File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.java
class Solution {
    public String reverseParentheses(String s) {
        Stack<Character> stack = new Stack<>();
        StringBuilder res = new StringBuilder();
        for(int i=0;i<s.length();i++) {
            char ch = s.charAt(i);
            // Opening bracket.
            if(ch == '(' || ch != ')') {
                stack.push(ch);
            }else {
                // Closing bracket.
                ArrayList<Character> list = new ArrayList<>();
                while(!stack.isEmpty() && stack.peek() != '(') {
                    list.add(stack.pop());
                }
                // Remove the opening bracket.
                stack.pop();
                for(char c: list) {
                    stack.push(c);
                }
            }
        }
        while(!stack.isEmpty()) {
            res.insert(0,stack.pop());
        }
        return res.toString();
    }
}

File: 1190-reverse-substrings-between-each-pair-of-parentheses/1190-reverse-substrings-between-each-pair-of-parentheses.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseParentheses = function(s) {
    const stack = [];
    let res = '';
    for(const ch of s) {
        if(ch === '(' || ch !== ')') {
            stack.push(ch);
        }
        else {
            const list = [];
            while(stack.length > 0 && stack[stack.length-1] !== '(') {
                list.push(stack.pop());
            }
            stack.pop();
            for(const c of list) {
                stack.push(c);
            }
        }
    }
    while(stack.length > 0) {
        res = stack.pop() + res;
    }
    return res;
};